### ISSUE TITLE
Decompiler: static property manual null assignment can still false-raise to ??=

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1845]
## Finding

While resolving #1746, adversarial review found the same false-positive class
still exists for static properties.

The #1746 fix narrows receiver/indexer property `??=` raises to require the
compiler value-spill discriminator. Static properties are intentionally left on
the pre-existing direct-setter path because, at `NullCoalescingAssignmentPass`
time, earlier expression inlining can erase the raw static `??=` spill before the
pass sees the `StoreProperty`.

## Suspected repro

```csharp
public static string? StaticProp { get; set; }

public static string ManualStaticPropertyNullAssignment(string fallback)
{
    if (StaticProp is null)
        StaticProp = fallback;
    return StaticProp;
}
```

Expected: manual static property null assignment should stay as an `if` unless
the pipeline preserves enough compiler-spill evidence to prove source `??=`.

Actual today: the no-receiver/no-index direct setter shape can still raise to
`StaticProp ??= fallback;`, preserving a latent opcode-diff risk.

## Why separate from #1746

#1746's concrete repro and fix cover instance property and indexer forms, where
the receiver/index place collapse is the bug and the value-spill discriminator is
still visible. Static properties need a different discriminator because
expression inlining can remove the spill before `NullCoalescingAssignmentPass`
runs.

## Acceptance

- Add compiled-fixture positive/negative coverage for true static property `??=`
  vs manual static property null assignment.
- Preserve true static `??=` altitude when the compiler discriminator is
  available, or decline honestly if it is not.
- Keep #1746's instance property/indexer positives and manual negatives green.

**Comment by richlander** [comment]
Curator update: added to semantic-wrong-`Full` burndown #1859 as an open runner row.
